lib/sign.js

```diff
@@ -37,25 +40,41 @@ function sign (manifest, program, options, config) {
       console.error();
       return reject();
     }
+    cmd.validateProgram(program);
 
-    resolve(xpi(manifest, options));
+    resolve(config.getManifest(options.xpi).then(function(manifest) {
+      var xpiConfig = {
+        manifest: manifest,
+      };
+      if (options.xpi) {
+        logger.log("Signing XPI: " + options.xpi);
+        return _.assign(xpiConfig, {xpiPath: options.xpi});
+      } else {
+        return xpi(manifest, options).then(function(xpiPath) {
+          logger.log("Created XPI for signing: " + xpiPath);
+          return _.assign(xpiConfig, {xpiPath: xpiPath});
```

Comment:
Seems like you could just call it `path` if you wanted since it's already about an XPI.